torch/_dynamo/variables/tensor.py

```diff
@@ -173,6 +175,31 @@ def specialize(value: torch.Tensor):
         return props
 
     def dynamic_getattr(self, tx, name):
+        fake_val = self.proxy.node.meta["example_value"]
+        # For getattrs on tensors without sources,
+        # we can do better than the default (creating a GetAttrVariable)
+        # if:
+        # (1) the tensor is a traceable tensor subclass
+        # (2) We are getattr'ing an inner tensor from that subclass
+        if not self.source and is_traceable_wrapper_subclass(fake_val):
+            fake_val = self.proxy.node.meta["example_value"]
+            attrs, ctx = fake_val.__tensor_flatten__()
+            if name in attrs or name in ctx:
+                proxy = getattr(self.as_proxy(), name)
+                example_value = getattr(fake_val, name)
+                if name in attrs:
+                    # attrs returned from tensor_flatten are always tensors
+                    assert isinstance(example_value, torch.Tensor)
+                    from .builder import wrap_fx_proxy
+
+                    return wrap_fx_proxy(
+                        tx=tx, proxy=proxy, example_value=example_value
+                    )
+                else:
+                    # attributes in the ctx returned by tensor_flatten are assumed to be constants
+                    from . import ConstantVariable
+
+                    return ConstantVariable(example_value)
```

Comment:
can ctx contain lists? lists cannot be ConstantVariable

Reply:
This probably needs some assertions that ctx is just what we think it is (ints? strings? etc)

Reply:
Actually, can we just send this through builder.py?